vector/view.go

```diff
@@ -38,6 +42,22 @@ func NewView(index []uint32, val Any) Any {
 	return &View{val, index}
 }
 
+func nullsView(nulls *Bool, index []uint32) *Bool {
+	if nulls == nil {
+		return nil
+	}
+	out := NewBoolEmpty(uint32(len(index)), nil)
+	for k, slot := range index {
+		if nulls.Value(slot) {
+			out.Set(uint32(k))
+		}
+	}
+	if out.Len() > 0 {
+		return out
+	}
+	return nil
```

Comment:
Nit: Maybe allocate a vector.Bool only when necessary?
```suggestion
	var out *vector.Bool
	for k, slot := range index {
		if nulls.Value(slot) {
			if out == nil {
				out = NewBoolEmpty(uint32(len(index)), nil)
			}
			out.Set(uint32(k))
		}
	}
	return out
```